src/main/java/org/example/bookstore/controller/BookControllerImpl.java

```diff
@@ -31,42 +32,48 @@ public class BookControllerImpl {
     private final BookService bookService;
 
     @GetMapping
+    @PreAuthorize("hasRole('ROLE_USER') or hasRole('ROLE_ADMIN')")
     @Operation(summary = "Get all books", description = "Get all existing books from DB")
     public List<BookResponseDto> getAllBooks(Pageable pageable) {
         return bookService.findAll(pageable);
     }
 
     @GetMapping("/{id}")
+    @PreAuthorize("hasRole('ROLE_USER') or hasRole('ROLE_ADMIN')")
     @Operation(summary = "Get book by id", description = "Get existing book by id")
     public BookResponseDto getBookById(@PathVariable Long id) {
         return bookService.getBookById(id);
     }
 
     @PostMapping
     @ResponseStatus(HttpStatus.CREATED)
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
```

Comment:
we need to add `ROLE_` when we use `hasAuthorithy()` but here we already specified that we are checking exactly role.
Same for other similar cases
```suggestion
    @PreAuthorize("hasRole('ADMIN')")
```